python-frontend/src/test/java/org/sonar/python/types/TypeInferenceTest.java

```diff
@@ -723,4 +729,36 @@ void user_defined_attributes_reassigned() {
     ).type()).isEqualTo(DECL_INT);
   }
 
+  @Test
+  void decorators() {
+    Decorator decorator = lastDecorator(
+      "class A:",
+      "  def dec_method():",
+      "    ...",
+      "my_dec = A()",
+      "@my_dec.dec_method()",
+      "def a_function():",
+      "  ...");
+    CallExpression ce = (CallExpression) decorator.expression();
+    QualifiedExpression qe = (QualifiedExpression) ce.callee();
+    assertThat(typeName(qe.qualifier().type())).isEqualTo("A");
+    assertThat(qe.name().symbol().fullyQualifiedName()).isEqualTo("package.mod.A.dec_method");
+    assertThat(ce.calleeSymbol().fullyQualifiedName()).isEqualTo("package.mod.A.dec_method");
+
+    decorator = lastDecorator(
+      "class A:",
+      "  def __call__():",
+      "    ...",
+      "@A",
+      "class OtherClass:",
+      "  ...");
+    Name name = (Name) decorator.expression();
+    assertThat(name.type()).isEqualTo(anyType());
```

Comment:
Not exactly sure which type `@A` should be here. I would think for now `Any` is fine.

Reply:
I think `Any` is fair here.

Currently, classes and functions are distinguished at symbol level and no type is defined for them. I think we eventually should infer types `type` and `Callable` (or something of the kind) for such symbols, but this is out of the scope of this ticket.